java/src/main/java/vowpalWabbit/learner/VWBase.java

```diff
@@ -96,4 +101,38 @@ public boolean equals(Object o) {
     public int hashCode() {
         return (int) (nativePointer ^ (nativePointer >>> 32));
     }
+
+    private class Closer implements Callable<Boolean> {
+        private Closer() {}
+
+        /**
+         * <p>
+         * Close the underlying VW model.
+         * </p>
+         *
+         * <p>
+         * May throw an unchecked exception if the underlying native code throws an Exception.
+         * </p>
+         * @return <code>true</code> if model was open and an attempt is made to close the model.
+         *         A return value of <code>false</code> indicates an attempt to close the model was
+         *         made previously.  This is consistent with other languages where a <code>true</code>
+         *         return value indicates success.
+         */
+        @Override
+        public Boolean call() {
+            lock.lock();
+            try {
+                final boolean attemptingToClose = isOpen;
+                if (isOpen) {
+                    isOpen = false;
+                    VWLearners.performRemainingPasses(nativePointer);
```

Comment:
I am a little nervous here if the C throws, `isOpen` is already set to `false` and there's no way to recover.  Should we wrap the native methods in a *try / catch* and in the catch reset `isOpen` to true, then rethrow the caught exception?